sympy/matrices/expressions/blockmatrix.py

```diff
@@ -556,38 +556,71 @@ def blockinverse_1x1(expr):
         return BlockMatrix(mat)
     return expr
 
+
 def blockinverse_2x2(expr):
-    if isinstance(expr.arg, BlockMatrix) and expr.arg.blockshape == (2, 2) and expr.arg.blocks[0, 0].is_square:
-        # Cite: The Matrix Cookbook Section 9.1.3
+    if isinstance(expr.arg, BlockMatrix) and expr.arg.blockshape == (2, 2):
+        # See: Inverses of 2x2 Block Matrices, Tzon-Tzer Lu and Sheng-Hua Shiou
         [[A, B],
          [C, D]] = expr.arg.blocks.tolist()
 
-        # Use one or the other formula, depending on whether A or D is known to be invertible or at least not known
-        # to not be invertible.  Note that invertAbility of the other expressions M is not checked.
-        A_invertible = ask(Q.invertible(A))
-        D_invertible = ask(Q.invertible(D))
-        if A_invertible == True:
-            invert_A = True
-        elif D_invertible == True:
-            invert_A = False
-        elif A_invertible != False:
-            invert_A = True
-        elif D_invertible != False:
-            invert_A = False
-        else:
-            invert_A = True
+        formula = _choose_2x2_inversion_formula(A, B, C, D)
```

Comment:
Should we let the user specify this as well via keyword argument? This seems to pick the D formula by default if nothing is known about the blocks.